### PR TITLE
RDMO 2.2.0

### PR DESCRIPTION
**Pull request by jochenklar**
* Add new projects overview (#865, #355)
  * Projects are now displayed in an interactive table which can be filtered and ordered
  * The projects API is now paginated (new setting PROJECT_TABLE_PAGE_SIZE)
* Add new import interface to management (#469, #468, #465, )
  * Show detailed information what is new and what changed
  * Show a summary of warnings and errors at the top of the page
* Add validation depending on the `value_type` configured for the question
  * Validation needs to be enabled using `PROJECT_VALUES_VALIDATION = True`
  * Configuration can be adjusted using settings for each value_type
* Enable markdown rendering for titles and texts of elements
* All parent attributes are now added to the full XML export
* Use only available catalogs for project import by users (#455)
* Add workaround for conflict validation for checkboxes (#903)
* Add `merge_attributes` management script to move related items from one attribute to another (#990)
* Add `join_values_inline` tag for views (#964)
* Add `user` and `site` to optionset provider plugins (#430)
* Add short title field for sections and pages for the navigation (#346, #363)
* Add section progress to the navigation
* Add button to add/remove the current site to an element in the management interface for multi site instances (#825)
* Fix a bug with conditions with non-consecutive set_index (when datasets are created and deleted)
* Fix a bug with the progress bar when a section has no pages
* Fix progress action if progress did not change
* Fix bugs with element copy in management (#995, #980)
* Fix typos and missing translations on buttons in management interface (#1020, #944)
* Fix link target for links in management interface (#1007)
* Fix textarea resizing (#1021)
* Fix export links in management (#915)
* Fix typos (#1001)
* Remove local hosts from ALLOWED_HOSTS settings
* Improve admin interface (#942, #918)
* Update default home page

### CODE REVIEW

**Comment by jochenklar** [comment]
Rebased to `2.1.1`.

**Comment by MyPyDavid** [comment]
rebased to 2.1.2 at https://github.com/rdmorganiser/rdmo/pull/892/commits/2dbdb49004ba05e66ef7d935eef99b61b62dbb54

**Comment by MyPyDavid** [comment]
rebased to main at 14f94ed86f331241ae0e911c0104cd1dad871782 but a test is failing

**Comment by MyPyDavid** [comment]
needed to rebase on main because of changes to `rdmo/core/templates/core/home_text_en.html`

**Comment by jochenklar** [comment]
Rebased and force-pushed to remove the double commits.

**Comment by MyPyDavid** [comment]
thanks for the cherry pick 🙏, should we set this PR to "ready for review" so that potential conflicts to main already become visible?

**Comment by MyPyDavid** [comment]
screenshots after merge and tests of import feature look fine 👍

**Comment by MyPyDavid** [comment]
I get this warning message in the browser when I browse `/projects/`:

```js
Warning: Modal uses the legacy childContextTypes API which is no longer supported and
will be removed in the next major release. 
Use React.createContext() instead

.Learn more about this warning here: https://reactjs.org/link/legacy-context
Modal@webpack://rdmo/./node_modules/react-bootstrap/es/Modal.js?:197:30
Modal@webpack://rdmo/./rdmo/core/assets/js/components/Modal.js?:15:16
div
Projects@webpack://rdmo/./rdmo/projects/assets/js/components/main/Projects.js?:25:16
Main@webpack://rdmo/./rdmo/projects/assets/js/containers/Main.js?:24:16
ConnectFunction@webpack://rdmo/./node_modules/react-redux/es/components/connect.js?:240:103
Provider@webpack://rdmo/./node_modules/react-redux/es/components/Provider.js?:15:18
DndProvider@webpack://rdmo/./node_modules/react-dnd/dist/core/DndProvider.js?:43:25 react-dom.development.js:86:30

``` 
Does anybody else also see this message or is it my local setup?
Maybe @CalamityC could look at this? It has to do with the Modal.

**Comment by jochenklar** [comment]
This is probably because of react-bootstrap, which we will get rid of eventually. I think its fine. We just need to stick to this React version for now.

File: rdmo/management/management/commands/merge_attributes.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    def handle(self, *args, **options):
+        source = options['source']
+        target = options['target']
+        save_changes = options['save']
+        delete_source = options['delete']
+        update_views = options['view']
+        verbosity = options.get('verbosity', 1)
+
+        source = get_valid_attribute(source, message_name='Source')
+        target = get_valid_attribute(target, message_name='Target')
```
Comment (MyPyDavid) [inline rdmo/management/management/commands/merge_attributes.py:58]: we ran into minor issue with this command. Currently, the target attribute can not be an attribute that has descendants but this condition matters only for the source and not for the target. 
I would like to change the `get_valid_attribute` function and only check the `attribute.get_descendants()` when eg. `allow_descendants` is not true

Comment (jochenklar) [reply rdmo/management/management/commands/merge_attributes.py:58]: go for it.

**Comment by MyPyDavid** [comment]
I found a minor issue with icon colors in the management.
Are these colors because of my faulty css variables or because of some missing css classes in the react components? 🤔 
The green is my link-color and the dark is the visited color, but I would expect that all the icons get the same color combination right? So that those blue icons would also be green.
image

**Comment by jochenklar** [comment]
I will check.

**Comment by MyPyDavid** [comment]
I could look at it tomorrow as well, the leftover PRs are ready for review https://github.com/rdmorganiser/rdmo/pulls?q=is%3Aopen+is%3Apr+milestone%3A%22RDMO+2.2.0%22

**Review by MyPyDavid — APPROVED** [review]
nice release with many changes! 

Test coverage could be improved afterwards
> Coverage decreased (-1.1%) to 90.467%

**Comment by jochenklar** [comment]
I fixed a tiny race condition in the `configureStore` for `projects`.

**Pull request merged**